Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/ActionInspectorContextMenu.cs

```diff
@@ -0,0 +1,171 @@
+            var hasSelection = m_ActionMapsTree.HasSelection();
+            var canAddBinding = false;
+            var action = m_ActionsTree.GetSelectedAction();
+            if (action != null && hasSelection)
+            {
+                canAddBinding = true;
+            }
+            if (canAddBinding)
+            {
+                menu.AddItem(isContextMenu ? m_AddBindingContextGUI : m_AddBindingGUI, false, OnAddBinding, treeViewItem);
+            }
+            else if (!isContextMenu)
+            {
+                menu.AddDisabledItem(m_AddBindingGUI);
+            }
+
+            var compositeString = isContextMenu ? EditorGUIUtility.TrTextContent("Create/Composite") : EditorGUIUtility.TrTextContent("Create Composite");
+            if (canAddBinding)
+            {
+                foreach (var composite in InputBindingComposite.s_Composites.names)
+                {
+                    menu.AddItem(new GUIContent(compositeString.text + " " + composite), false, OnAddCompositeBinding, new object[]{treeViewItem, composite});
+                }
+            }
+            else if (!isContextMenu)
+            {
+                menu.AddDisabledItem(new GUIContent(compositeString), false);
+            }
+        }
+
+        private void OnAddCompositeBinding(object objs)
+        {
+            var actionLine = ((object[])objs)[0] as ActionTreeItem;
+            var compositeName = ((object[])objs)[1] as string;
+            if (actionLine == null)
+                return;
+            actionLine.AddCompositeBinding((string)compositeName);
+            m_AssetInspectorWindow.Apply();
+        }
+
+        private void OnAddBinding(object actionLineObj)
+        {
+            var actionLine = actionLineObj as ActionTreeItem;
+            if (actionLine == null)
+                return;
+            actionLine.AddBinding();
+            m_AssetInspectorWindow.Apply();
+        }
+
+        public void OnAddAction()
+        {
+            var actionMapLine = GetSelectedActionMapLine();
+            if (actionMapLine == null)
+                return;
+            actionMapLine.AddAction();
+            m_AssetInspectorWindow.Apply();
+        }
+
+        public void OnAddActionMap()
+        {
+            InputActionSerializationHelpers.AddActionMap(m_ActionAssetManager.serializedObject);
+            m_AssetInspectorWindow.Apply();
+        }
+
+        private ActionTreeItem GetSelectedActionLine()
+        {
+            TreeViewItem selectedRow = m_ActionsTree.GetSelectedRow();
+            do
+            {
+                if (selectedRow is ActionTreeItem)
+                    return (ActionTreeItem)selectedRow;
+                selectedRow = selectedRow.parent;
+            }
+            while (selectedRow.parent != null);
+
+            return null;
+        }
+
+        private ActionMapTreeItem GetSelectedActionMapLine()
+        {
+            TreeViewItem selectedRow = m_ActionMapsTree.GetSelectedRow();
```

Comment:
Please use `var` for all local variables.

Reply:
It needs to be specific type in this case.